Sources/tasks/CreatePost-Notify.php

```diff
@@ -182,10 +195,6 @@ public function execute()
 		// Modified post, or dealing with delayed mention and quote notifications.
 		if ($type == 'edit' || !empty($this->_details['respawns']))
 		{
-			// Filter out members who have already been notified about this post's topic
-			$this->members['quoted'] = array_intersect_key($this->members['quoted'], $unnotified);
-			$this->members['mentioned'] = array_intersect_key($this->members['mentioned'], $unnotified);
-
```

Comment:
This logic is broken, yes. However, just deleting it will cause two notifications to be sent members who get mentioned in a topic they are watching, so we need to fix the logic here rather than remove it.

I suggest the following:

```
// Filter out members who have already been notified about this post's topic
$already_notified = array_diff_key($this->members['watching'], $unnotified);
$this->members['quoted'] = array_diff_key($this->members['quoted'], $already_notified);
$this->members['mentioned'] = array_diff_key($this->members['mentioned'], $already_notified);
```

Reply:
Thanks, I'll take another look at this once we're past all the 2.1.0 stuff!

Reply:
Actually it already keeps track of who has already received a notification in $this->members['done'], and will not send another notice to folks in the 'done' list.  I have not been able to get it to send dupes in test.

I think we're good!

https://github.com/SimpleMachines/SMF2.1/blob/54b854bba9abddf1f351ab48eed0edc347027366/Sources/tasks/CreatePost-Notify.php#L464
https://github.com/SimpleMachines/SMF2.1/blob/54b854bba9abddf1f351ab48eed0edc347027366/Sources/tasks/CreatePost-Notify.php#L632
https://github.com/SimpleMachines/SMF2.1/blob/54b854bba9abddf1f351ab48eed0edc347027366/Sources/tasks/CreatePost-Notify.php#L706

Reply:
Fair enough then. 🙂